src/main/java/com/fasterxml/jackson/databind/jsontype/impl/AsPropertyTypeDeserializer.java

```diff
@@ -182,7 +202,9 @@ protected Object _deserializeTypedUsingDefaultImpl(JsonParser p,
         // genuine, or faked for "dont fail on bad type id")
         JsonDeserializer<Object> deser = _findDefaultImplDeserializer(ctxt);
         if (deser == null) {
-            JavaType t = _handleMissingTypeId(ctxt, priorFailureMsg);
+            JavaType t = _hasTypeResolver
+                ? _handleMissingTypeId(ctxt, priorFailureMsg) : ctxt.constructType(_baseType.getRawClass());
```

Comment:
Shouldn't this just be

    ? _handleMissingTypeId(ctxt, priorFailureMsg) : _baseType

so as not to lose generic types possible present in base (or in general just to avoid extra work)?

Reply:
I can try it out, after merging.

Reply:
Ok perfect, thank you!